opslevel/terraform_type_conversions.go

```diff
@@ -117,6 +117,15 @@ func StringSliceToSetValue(values []string) basetypes.SetValue {
 	return types.SetValueMust(types.StringType, result)
 }
 
+// Converts a []string to a basetypes.SetValue
+func StringSliceToListValue(values []string) basetypes.ListValue {
```

Comment:
This looks like a duplicate of `OptionalStringListValue()`

`func StringSliceToListValue(values []string) basetypes.ListValue { ... } `
`func OptionalStringListValue(values []string) basetypes.ListValue { ... }`

Reply:
@davidbloss this function is still necessary here because we don't want this to return `types.ListNull(types.StringType)`

Reply:
This is definitely a one off. I'm looking around and only see this function being called once in `newIntegrationGoogleCloudResourceModel()` for `ownership_tag_keys` which has a Default value set. 

The docs for `field Default defaults.List` read:
```tf
Default defines a proposed new state (plan) value for the attribute if the configuration value is null. 
Default prevents the framework from automatically marking the value as unknown during planning when other proposed new state changes are detected. 
If the attribute is computed and the value could be altered by other changes then a default should be avoided and a plan modifier should be used instead.
```

This leads me to believe that if a TF config omits `ownership_tag_keys` or if it's set to `ownership_tag_keys = null` then the `Default:` will propose `["owner"]` and there wouldn't be a way it could be set to `null`

Reply:
I am confused here, my rationale is that `OptionalStringListValue` that already exists has this as its first statement,

```
if len(values) == 0 {
		return types.ListNull(types.StringType)
	}
```

Meaning that the user setting the ownership tag keys to `[]` would not work if using that function which is why `StringSliceToListValue` is necessary (can never return a null list, only empty list)

We want the user to be able to use `[]` in addition to just not setting ownership tag keys (in that case it would default to `["owner"]` but we still need to support `[]`).